src/core/http/request_context.rs

```diff
@@ -263,14 +263,38 @@ mod test {
     }
 
     #[test]
-    fn test_is_batching_enabled_default() {
+    fn test_is_batching_disabled_default() {
         // create ctx with default batch
         let config_module = config::ConfigModule::default();
         let mut upstream = Upstream::try_from(&config_module).unwrap();
         let server = Server::try_from(config_module).unwrap();
         upstream.batch = Some(Batch::default());
         let req_ctx: RequestContext = RequestContext::default().upstream(upstream).server(server);
 
-        assert!(req_ctx.is_batching_enabled());
+        assert!(!req_ctx.is_batching_enabled());
+    }
+
+    #[test]
+    fn test_is_batching_disabled_for_delay_zero() {
+        // create ctx with default batch
+        let config_module = config::ConfigModule::default();
+        let mut upstream = Upstream::try_from(&config_module).unwrap();
+        let server = Server::try_from(config_module).unwrap();
+        upstream.batch = Some(Batch { delay: 0, ..Default::default() });
+        let req_ctx: RequestContext = RequestContext::default().upstream(upstream).server(server);
+
+        assert!(!req_ctx.is_batching_enabled());
+    }
+
+    #[test]
+    fn test_is_batching_disabled_for_max_size_none() {
+        // create ctx with default batch
+        let config_module = config::ConfigModule::default();
+        let mut upstream = Upstream::try_from(&config_module).unwrap();
+        let server = Server::try_from(config_module).unwrap();
+        upstream.batch = Some(Batch { max_size: None, ..Default::default() });
+        let req_ctx: RequestContext = RequestContext::default().upstream(upstream).server(server);
```

Comment:
Lot of duplicate code in the tests. Can we reduce it and move some common parts to a function.

Reply:
created a separate function `create_req_ctx_with_batch` for creating the `RequestContext` object with the specified `Batch` object